application/src/test/java/org/thingsboard/server/controller/AbstractWebTest.java

```diff
@@ -699,4 +711,49 @@ protected <T extends HasId<? extends UUIDBased>> ListenableFuture<List<ResultAct
         return Futures.allAsList(futures);
     }
 
+    protected void tesEntityDaoWithRelationsOk(EntityId entityIdFrom, EntityId entityTo, String urlDelete) throws Exception {
+        createEntityRelation(entityIdFrom, entityTo, "TEST_TYPE");
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+
+        doDelete(urlDelete).andExpect(status().isOk());
+
+        assertThat(getRelationsTo(entityTo)).hasSize(0);
+    }
+
+    protected <T> void tesEntityDaoWithRelationsTransactionalException(Dao<T> dao, EntityId entityIdFrom, EntityId entityTo,
+                                                                       String urlDelete) throws Exception {
+        entityDaoRemoveByIdWithException (dao);
+        createEntityRelation(entityIdFrom, entityTo, "TEST_TRANSACTIONAL_TYPE");
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+
+        doDelete(urlDelete)
+                .andExpect(status().isInternalServerError());
+
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+    }
+
+    protected <T> void entityDaoRemoveByIdWithException (Dao<T> dao) {
+        BDDMockito.willThrow(new ConstraintViolationException("mock message", new SQLException(), "MOCK_CONSTRAINT"))
+                .given(dao).removeById(any(), any());
+    }
+
+    protected <T> void afterTestEntityDaoRemoveByIdWithException (Dao<T> dao) {
+        BDDMockito.willCallRealMethod().given(dao).removeById(any(), any());
+    }
+
+    protected void createEntityRelation(EntityId entityIdFrom, EntityId entityIdTo, String typeRelation) throws Exception {
+        EntityRelation relation = new EntityRelation(entityIdFrom, entityIdTo, typeRelation);
+        doPost("/api/relation", relation);
+    }
+
+    protected List<EntityRelation> getRelationsTo(EntityId entityId) throws Exception {
```

Comment:
rename 'findRelationsByTo'. Same naming as used in relations service

Reply:
OK